### PR TITLE
Update BlueSCSI.cpp

### PR DESCRIPTION
**Pull request by thegouldfish**
Updating the command reading to be similar to RaScsi.

### CODE REVIEW

File: src/BlueSCSI.cpp [hunk 1]
```diff
@@ -1230,25 +1230,33 @@ void loop()
...
   byte cmd[12];
-  cmd[0] = readHandshake(); if(m_isBusReset) goto BusFree;
+  int cmdCount;
+  cmd[0] = readHandshake(); 
+  if(m_isBusReset) 
+    goto BusFree;
+
   LOGHEX(cmd[0]);
-  // Command length selection, reception
-  static const int cmd_class_len[8]={6,10,10,6,6,12,6,6};
```
Comment (erichelgeson) [inline src/BlueSCSI.cpp:1238]: Looks like one command has 12 - or it could be a typo/bug.

Comment (thegouldfish) [reply src/BlueSCSI.cpp:1238]: I'm not sure, this code was based on what I found in the raSCSI code base.
https://github.com/akuker/RASCSI/blob/master/src/raspberrypi/gpiobus.cpp "GPIOBUS::CommandHandShake"
with this change it was more stable on the Amiga (not 100% but better then with out it).
Its possible that BlueSCSI is implementing a command that raSCSI doesn't or that there is a mistake in blue or in ra.
its hard to tell.

Comment (thegouldfish) [reply src/BlueSCSI.cpp:1238]: Looking up some documention it seems both might be wrong in some way.
as there is a 16byte command as well.
though looking at the code none of the implemented commands seem to use more then the first 8 bytes.
image
https://www.seagate.com/files/staticfiles/support/docs/manual/Interface%20manuals/100293068j.pdf

Comment (thegouldfish) [reply src/BlueSCSI.cpp:1238]: Its actually what I've done in an experimental branch and it seemed fine.
I've been checking out a bunch of other STM32 based scsi source codes and most only implement 3 group codes.
so this having all of them is overkill, but doesn't harm the code.

**Comment by erichelgeson** [comment]
@thegouldfish - could you try the recent changes to main branch? There was some changes that may resolve your original issue we discussed in slack. I'm going to close this as main has diverged, but if you find something please let us know.

**Pull request closed without merging**